diff -Naur -x '*.o*' drivers/mfd.orig/mc-pmic-core.c drivers/mfd/mc-pmic-core.c
--- a/drivers/mfd/mc-pmic-core.c	2012-05-07 16:07:14.000000000 +0300
+++ b/drivers/mfd/mc-pmic-core.c	2012-05-07 16:07:21.000000000 +0300
@@ -505,7 +505,7 @@
 
 	struct mfd_cell cell = {
 		.platform_data = pdata,
-		.data_size = pdata_size,
+		.pdata_size = pdata_size,
 	};
 
 	/* there is no asnprintf in the kernel :-( */
diff -Naur -x '*.o*' drivers/mfd.orig/da9052-core.c drivers/mfd/da9052-core.c
--- a/drivers/mfd/da9052-core.c	2012-05-07 16:32:10.000000000 +0300
+++ b/drivers/mfd/da9052-core.c	2012-05-07 16:32:28.000000000 +0300
@@ -376,7 +376,7 @@
 	struct mfd_cell cell = {
 		.name = name,
 		.platform_data = pdata,
-		.data_size = pdata_size,
+		.pdata_size = pdata_size,
 	};
 	return mfd_add_devices(da9052->dev, -1, &cell, 1, NULL, 0);
 }
